src/libraries/Enableable.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {RolesConsumer} from "src/modules/ROLES/OlympusRoles.sol";
+
+/// @title  Enableable
+/// @notice This contract is designed to be inherited by contracts that need to be enabled or disabled. It replaces the inconsistent usage of `active` and `locallyActive` state variables across the codebase.
+/// @dev    A contract that inherits from this contract should use the `whenEnabled` and `whenDisabled` modifiers to gate access to certain functions.
+///
+///         If custom logic and/or parameters are needed for the enable/disable functions, the inheriting contract can override the `_enable()` and `_disable()` functions. For example, `enable()` could be called with initialisation data that is decoded, validated and assigned in `_enable()`.
+abstract contract Enableable is RolesConsumer {
+    // ===== STATE VARIABLES ===== //
+
+    bool public isEnabled;
+    bytes32 public constant ROLE = "emergency_shutdown";
+
+    // ===== ERRORS ===== //
+
+    error NotDisabled();
+    error NotEnabled();
+
+    // ===== EVENTS ===== //
+
+    event Disabled();
+    event Enabled();
+
+    // ===== MODIFIERS ===== //
+
+    modifier whenEnabled() {
+        if (!isEnabled) revert NotEnabled();
+        _;
+    }
+
+    modifier whenDisabled() {
+        if (isEnabled) revert NotDisabled();
+        _;
+    }
+
+    // ===== ENABLEABLE FUNCTIONS ===== //
+
+    /// @notice Enable the contract
+    /// @dev    This function performs the following steps:
+    ///         1. Validates that the caller has `ROLE` ("emergency_shutdown")
```

Comment:
A general question -- does it make sense for this to be the `emergency_shutdown` role only? 

For example, for a new policy deployment `isEnabled=false`, meaning you have to get the `emergency_shutdown` to first enable it. 

Is that expected or should it be some other (new) admin role, where the emergency msig and/or perhaps other addresses should be allowed?

@ohmzeus wdyt?

Reply:
That's a good suggestion. I think it would be good to have the enable/disable functions be accessible to the "emergency" AND "admin" roles. "emergency" will typically be quicker to respond, but it's kludgy during initial setup of a contract.